### DISCUSSION TITLE
How to make Rule when ibeacon is visible do Power1 1

### DISCUSSION THREAD

**Discussion by gbananachewicz** [arendst/Tasmota#16201] [Q&A]
How to make Rule when ibeacon (mac or name )and RSSI -70  is visible do Power1 1 

LOG :
16:31:18.927 RSL: SENSOR = {"Time":"2022-08-10T16:31:18","IBEACON":{"MAC":"FFFAC10FDEA2","NAME":"grzes-klucze","RSSI":0,"STATE":"OFF","PERSEC":0}}
I can't do it 
ESP32 v. D  Tasmota Bluetooth12.0.2.4

**Comment by sfromis** [comment]
On ESP32, I'd recommend that you instead use Berry, where such things are more straightforward to do.
```
tasmota.add_rule('IBEACON#NAME=grzes-klucze', 
  def (value,trigger,jsonmap)
    var rssi=jsonmap['IBEACON']['RSSI']
    if rssi > -70
      tasmota.set_power(0, true)
    end
  end)
```
Of course, you may need to adapt the condition `> -70` to work for your use case.

**Reply by gbananachewicz** [reply]
Thanks a lot. I have this code working, How do I make more IBeacons by name?

**Reply by sfromis** [reply]
Same way as the first one. When you have another beacon running, with `ZbName` assigned, you can simply have another `tasmota.add_rule` for that one.

(And do put your Berry init functions in `autoexec.be` if you want to have them also after next boot).

**Reply by sfromis** [reply]
Depending on details, it might also be reasonable to use a different trigger to have more than one beacon handled by the same add_rule.

**Reply by gbananachewicz** [reply]
tasmota.add_rule('IBEACON#NAME=grzes-klucze', || 
 IBEACON#NAME=grzes-klucze2',

How to add more IBeacons to rule ?

**Reply by barbudor** [reply]
You could trigger only on `IBEACON` and then process in the Berry function like:
```
tasmota.add_rule('IBEACON', 
  def (value,trigger,jsonmap)
    if jsonmap['IBEACON']['NAME'] == 'name1'
           ... do whatever for ibeacon name 1 ..
    elif jsonmap['IBEACON']['NAME'] == 'name2'
           ... do whatever for ibeacon name 1 ..
    else
         print("I don't know this iBeacon ", jsonmap['IBEACON']['NAME'] )
    end
  end)
```

**Reply by sfromis** [reply]
Or you can have multiple `add_rule` calls, as already explained. To avoid repeating the handler code, it could be a defined non-anonymous function used in each add__rule.

**Reply by gbananachewicz** [reply]
How to do tasmoya.delay(6000) because while delay time console logs and working of Esp32 hangs for this time. I need 3 to 6 seconds delay ?

**Reply by sfromis** [reply]
Do not use `tasmota.delay` unless below 50 milliseconds, Tasmota is blocked completely. If you need something to happen later, use `tasmota.set_timer`.

**Comment by gbananachewicz** [comment]
How to add Ibeacon UID to known beacons like Blealias ?
